app/controllers/ops_controller/settings/common.rb

```diff
@@ -218,7 +225,18 @@ def pglogical_save_subscriptions
         add_flash(_("Error during replication configuration save: %{message}") %
                     {:message => bang.message}, :error)
       else
-        add_flash(_("Replication configuration save was successful"))
+        if replication_type == :remote && !params[:exclusion_list].empty?
```

Comment:
I'm thinking this code should probably be in a separate method.

I don't think this really belongs in `pglogical_save_subscriptions` right?

I'm thinking the entry point should be a separate, more generic method. Then depending on the replication type we should call something like `pglogical_save_subscriptions` or `pglogical_save_excludes`. That would make this a lot easier to understand.

Then we could also remove some duplication of the error handling code and also of `MiqRegion.replication_type = replication_type` calls (which we seem to have two of).